src/react-components/ui-root.js

```diff
@@ -1356,10 +1382,42 @@ class UIRoot extends Component {
                     placeholder="Send a message..."
                   />
                   <input
-                    className={classNames([styles.messageEntrySubmit, styles.messageEntrySubmitInRoom])}
-                    type="submit"
-                    value="send"
+                    id="message-entry-media-input"
+                    type="file"
+                    style={{ display: "none" }}
+                    accept={isMobile ? "image/*" : "*"}
+                    multiple
+                    onChange={e => {
+                      for (const file of e.target.files) {
+                        this.createObject(file);
+                      }
+                    }}
                   />
+                  <label
+                    htmlFor="message-entry-media-input"
+                    title={"Upload"}
+                    className={classNames([
+                      styles.messageEntryButton,
+                      styles.messageEntryButtonInRoom,
+                      styles.messageEntryUpload
+                    ])}
+                  >
+                    <i>
+                      <FontAwesomeIcon icon={isMobile ? faCamera : faPlus} />
+                    </i>
+                  </label>
+                  <button
+                    type="submit"
+                    className={classNames([
+                      styles.messageEntryButton,
+                      styles.messageEntryButtonInRoom,
+                      styles.messageEntrySubmit
+                    ])}
+                  >
+                    <i>
+                      <FontAwesomeIcon icon={faPaperPlane} />
+                    </i>
+                  </button>
```

Comment:
Isn't it better to put the send button before the upload button? I would also suggest using the same cloud icon we use in the media browser.

![firefox_2019-02-19_16-19-34](https://user-images.githubusercontent.com/79419/53057324-cc916c80-3463-11e9-86b8-1e394b8476d0.png)

Reply:
actually now that I think about it, i should reorder these completely. the reason for the + was to match slack and discord. i think the right layout would be:

```
[+ Send a message [spawn] [send]]
```